### PR TITLE
adding id prop for summary text span tag

### PR DESCRIPTION
**Pull request by arpakuma**
Issue  - https://github.com/Capgemini/dcx-react-library/issues/438

### CODE REVIEW

File: src/details/Details.tsx [hunk 1]
```diff
@@ -30,6 +30,10 @@ type DetailsProps = {
    * summary class name
    */
   summaryClassName?: string;
+   /**
+   * summary text id name
+   */
+  summaryTextId?: any;
```
Comment (daniele-zurico) [inline src/details/Details.tsx:36]: as I suggested on the ticket please use summaryProps as property.
something like:
```js
summaryTextProps?:  React.AllHTMLAttributes<HTMLSpanElement>;
```
check the `FormInput` component as example (line 61 and 229)

**Review by Ibabalola — COMMENTED** [review]
Hi @arpakuma - please also help us to update the documentation with this new prop
`stories/Details/Documentation.stories.mdx` and `stories/liveEdit/DetailsLive.tsx`

File: src/details/Details.tsx (hunk 1 above)
Comment (arpakuma) [reply src/details/Details.tsx]: updated the property to summaryTextProps. Please review. Thanks!

**Comment by arpakuma** [comment]
> 

updated the documentation. Please review. Thanks!

File: src/details/Details.tsx [hunk 2]
```diff
@@ -57,7 +62,7 @@ export const Details = ({
   return (
     <details className={detailsClassName} open={open}>
       <summary className={summaryClassName} onClick={handleOnChange}>
-        <span className={summaryTextClassName}>{summary}</span>
+        <span id={summaryTextProps?.id} className={summaryTextClassName} {...summaryTextProps}>{summary}</span>
```
Comment (daniele-zurico) [inline src/details/Details.tsx:65]: please remove `id={summaryTextProps?.id}`

File: src/details/__test__/Details.test.tsx [hunk 3]
```diff
@@ -159,4 +159,24 @@ describe('Details', () => {
...
+    const { container } = render(
+      <Details
+        summary="my summary"
+        summaryTextClassName="summary-text-class-name"
+        summaryTextProps={{id:'summary-text-id'}}
+      >
+        my details
+      </Details>
+    );
+    expect(container.querySelector('.summary-text-class-name')).not.toBeNull();
```
Comment (daniele-zurico) [inline src/details/__test__/Details.test.tsx:173]: I don't thing you need from line 173 to 177

**Review by daniele-zurico — APPROVED** [review]

File: src/details/Details.tsx [hunk 4]
```diff
@@ -30,6 +30,10 @@ type DetailsProps = {
    * summary class name
    */
   summaryClassName?: string;
+   /**
+   * allow to customise the span with all the properites needed
```
Comment (daniele-zurico) [inline src/details/Details.tsx:34]: please fix the typo `properites`

**Review by daniele-zurico — APPROVED** [review]

**Review by Ibabalola — APPROVED** [review]

**Pull request merged**